public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ResourcePicker/utils.ts

```diff
@@ -1,35 +1,85 @@
 import produce from 'immer';
 
+import { getTemplateSrv } from '@grafana/runtime';
+
+import UrlBuilder from '../../azure_monitor/url_builder';
+import { AzureMetricResource, AzureMonitorQuery } from '../../types';
+
 import { ResourceRow, ResourceRowGroup } from './types';
 
 // This regex matches URIs representing:
 //  - subscriptions: /subscriptions/44693801-6ee6-49de-9b2d-9106972f9572
 //  - resource groups: /subscriptions/44693801-6ee6-49de-9b2d-9106972f9572/resourceGroups/cloud-datasources
 //  - resources: /subscriptions/44693801-6ee6-49de-9b2d-9106972f9572/resourceGroups/cloud-datasources/providers/Microsoft.Compute/virtualMachines/GithubTestDataVM
 const RESOURCE_URI_REGEX =
-  /\/subscriptions\/(?<subscriptionID>[^/]+)(?:\/resourceGroups\/(?<resourceGroup>[^/]+)(?:\/providers.+\/(?<resource>[^/]+))?)?/;
+  /\/subscriptions\/(?<subscription>[^/]+)(?:\/resourceGroups\/(?<resourceGroup>[^/]+)(?:\/providers\/(?<metricNamespaceAndResource>.+))?)?/;
 
 type RegexGroups = Record<string, string | undefined>;
 
+function parseNamespaceAndName(metricNamespaceAndName?: string) {
```

Comment:
Do we need to worry about badly formatted strings here?
What if the stringArray length is less than 2 , or if the remaining in loop is not an even number?

Would that break something elsewhere?

Reply:
That would result in a invalid resource (so it should not work today). Anyway, `.splice` won't fail, just return an empy array so it wouldn't crash. I have added test cases to cover these just in case.